src/main/java/core/basesyntax/dao/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.dp.Storage;
+import java.util.Map;
+import java.util.Set;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public void setQuantity(String fruitName, Integer quantity) {
+        Storage.storage.put(fruitName, quantity);
+    }
+
+    @Override
+    public Integer getQuantity(String fruitName) {
+        if (!Storage.storage.containsKey(fruitName)) {
+            throw new RuntimeException("There is no " + fruitName + " in the storage");
+        }
+        return Storage.storage.get(fruitName);
+    }
+
+    @Override
+    public Set<Map.Entry<String, Integer>> getAll() {
+        return Map.copyOf(Storage.storage).entrySet();
```

Comment:
The `getAll` method uses `Map.copyOf(Storage.storage).entrySet()`, which creates a copy of the storage map and then returns its entry set. This could be inefficient if the storage is large. Consider returning `Storage.storage.entrySet()` directly if immutability is not a concern, or use `Collections.unmodifiableSet()` to return an unmodifiable view if needed.